chemprop/uncertainty/calibrator.py

```diff
@@ -110,21 +112,77 @@ def apply(self, uncs: Tensor) -> Tensor:
 
 
 @UncertaintyCalibratorRegistry.register("conformal-regression")
-class ConformalRegressionCalibrator(RegressionCalibrator):
-    def fit(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -> Self:
-        return self
+class RegressionConformalCalibrator(RegressionCalibrator):
+    r"""Conformalize quantiles to make the interval :math:`[\hat{t}_{\alpha/2}(x),\hat{t}_{1-\alpha/2}(x)]` to have
+    approximately :math:`1-\alpha` coverage. [1]_
 
-    def apply(self, uncs: Tensor) -> Tensor:
-        return
+    .. math::
+        s(x, y) &= \max \left\{ \hat{t}_{\alpha/2}(x) - y, y - \hat{t}_{1-\alpha/2}(x) \right\}
+
+        \hat{q} &= Q(s_1, \ldots, s_n; \left\lceil \frac{(n+1)(1-\alpha)}{n} \right\rceil)
 
+        C(x) &= \left[ \hat{t}_{\alpha/2}(x) - \hat{q}, \hat{t}_{1-\alpha/2}(x) + \hat{q} \right]
+
+    where :math:`s` is the nonconformity score as the difference between y and its nearest quantile.
+    :math:`\hat{t}_{\alpha/2}(x)` and :math:`\hat{t}_{1-\alpha/2}(x)` are the predicted quantiles from quantile
```

Comment:
Does this method require the predicted quantiles from a quantile regression model (i.e. one trained with pinball loss)? It looks like it a single value prediction and a symmetric error bound.

Reply:
Yes, it still needs the predicted upper and lower quantiles. Conformal prediction tries to adjust the interval size to match the predefined coverage (i.e., $1 - \alpha$).